java/vortex-spark/src/main/java/dev/vortex/spark/read/VortexScanBuilder.java

```diff
@@ -50,18 +55,29 @@ public VortexScanBuilder addAllPaths(Iterable<String> paths) {
     }
 
     public VortexScanBuilder addAllColumns(Iterable<Column> columns) {
-        this.columns.addAll(columns);
+        for (Column column : columns) {
+            this.columns.add(column);
+        }
         return this;
     }
 
     @Override
     public Scan build() {
         var paths = this.paths.build();
-        var columns = this.columns.build();
+        var columns = ImmutableList.copyOf(this.columns);
 
         checkState(!paths.isEmpty(), "paths cannot be empty");
         checkState(!columns.isEmpty(), "readColumns cannot be empty");
 
         return new VortexScan(paths, columns);
     }
+
+    @Override
+    public void pruneColumns(StructType requiredSchema) {
+        // TODO(aduffy): support deeply nested schema prunes
```

Comment:
I don't follow this comment. It seems like the code below will ensure only (nested) fields explicitly in requiredSchema will be in columns. Am I missing something?

Reply:
It won't do deeply nested pruning, e.g. struct of struct of struct